monkey/monkey_island/cc/repository/mongo_event_repository.py

```diff
@@ -0,0 +1,71 @@
+from typing import Any, MutableMapping, Sequence, Type
+
+from pymongo import MongoClient
+
+from common.event_serializers import EVENT_TYPE_FIELD, EventSerializerRegistry
+from common.events import AbstractAgentEvent
+from common.types import AgentID
+from monkey_island.cc.repository import IEventRepository
+
+from . import RemovalError, RetrievalError, StorageError
+from .consts import MONGO_OBJECT_ID_KEY
+
+
+class MongoEventRepository(IEventRepository):
+    """A repository for storing and retrieving events in MongoDB."""
+
+    def __init__(self, mongo_client: MongoClient, serializer_registry: EventSerializerRegistry):
+        self._events_collection = mongo_client.monkey_island.events
+        self._serializers = serializer_registry
+
+    def save_event(self, event: AbstractAgentEvent):
+        try:
+            serializer = self._serializers[type(event)]
+            serialized_event = serializer.serialize(event)
+            self._events_collection.insert_one(serialized_event)
+        except Exception as err:
+            raise StorageError(err)
+
+    def get_events(self) -> Sequence[AbstractAgentEvent]:
+        try:
+            serialized_events = list(self._events_collection.find())
+            return list(map(self._deserialize, serialized_events))
```

Comment:
I think `serialized_events = list(...` is unnecessary. `map()` takes an `Iterable` and `find()` returns a `Cursor` which I believe is iterable.

Reply:
It returns an [`Iterator`](https://docs.python.org/3/library/functions.html#map)

Reply:
`map()` does. I'm saying `find()` does also, so you don't need to `list(self._events_collection.find())`.

Reply:
Aha! Gotcha